pkg/controller/dynamicrequiredmanifest_controller_test.go

```diff
@@ -0,0 +1,163 @@
+package controller
+
+import (
+	"context"
+	"errors"
+	"strings"
+	"testing"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+
+	catalogdv1alpha1 "github.com/operator-framework/catalogd/api/core/v1alpha1"
+)
+
+func containsError(expected error) func(t *testing.T, err error) {
+	return func(t *testing.T, err error) {
+		if err == nil || !strings.Contains(err.Error(), expected.Error()) {
+			t.Fatalf("expected an error with substring %q but received error %v", expected.Error(), err)
+		}
+	}
+}
+
+func noError() func(t *testing.T, err error) {
+	return func(t *testing.T, err error) {
+		if err != nil {
+			t.Fatalf("expected no error but received error %v", err)
+		}
+	}
+}
+
+func TestClusterCatalogControllerSync(t *testing.T) {
```

Comment:
Need to update this test name and any references to catalogs

Reply:
Done in https://github.com/openshift/cluster-olm-operator/pull/73/commits/1dd7f1fd8255a006a832402bb6d44726983373d3